### PR TITLE
Feature/web-components

### PR DESCRIPTION
**Pull request by Dashkent**

### CODE REVIEW

File: staff/daria-potemkina/web-components/my-app-react/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+const { useState, useEffect } = React
+
+function App() {
+
+    const [view, setView] = useState('load')
+    // const [name, setName] = useState()
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/components/App.jsx:6]: remember to remove comments of code that you don't use

File: staff/daria-potemkina/web-components/my-app-react/components/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    handleGoToUsers = (event) => {
+        event.preventDefault()
+
+        this.goToView('users')
+    }
+
+    handleSearchUsers = query => {
+        searchUsers(this.props.token, query, (error, users) => {
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/components/Home.jsx:58]: Remebr that you always need to wrap logic functions in a try catch for handling synchronous errors.

File: staff/daria-potemkina/web-components/my-app-react/components/Home.jsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    handleGoToGoogle = (event) => {
+        event.preventDefault()
+
+        this.goToView('google')
+    }
+
+    handleGoogle = query => {
+        searchGoogle(query, (error, results) => {
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/components/Home.jsx:71]: same here

File: staff/daria-potemkina/web-components/my-app-react/components/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    handleGoToEcosia = event => {
+        event.preventDefault()
+
+        this.goToView('ecosia')
+    }
+
+    handleEcosia = query => {
+        searchEcosia(query, (error, results) => {
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/components/Home.jsx:84]: and here

File: staff/daria-potemkina/web-components/my-app-react/components/Home.jsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    handleGoToLogout = event => {
+        event.preventDefault();
+
+        this.props.onLogout()
+    }
+
+    handleFollowing = id => {
+        toggleFollowUser(this.props.token, id, error => {
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/components/Home.jsx:103]: and here

File: staff/daria-potemkina/web-components/my-app-react/components/Twitter.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
+const { useState, useEffect} = React
+
+function Twitter ({token}) {
+  
+     const [error, setError] = useState()
+     const [tweets, setTweets] = useState()
+    
+    useEffect(() => {
+        retrieveTweets(token, (error, allTweets) => {
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/components/Twitter.jsx:9]: again here remember that you need to wrap everything in try catch

File: staff/daria-potemkina/web-components/my-app-react/components/Twitter.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            debugger
+            if (error) throw error
+            setTweets(allTweets)
+
+        })
+    },[])
+    
+     const handleSubmitTweet = tweets => {
+        tweet(token, tweets, error => {
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/components/Twitter.jsx:18]: here too

File: staff/daria-potemkina/web-components/my-app-react/logic/is-user.authenticate.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+function isUserAuthenticate(token, callback){
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/logic/is-user.authenticate.js:1]: The name of the file shoud be `is-user-authenticated` with a `-` not `is-user.authenicated` with a `.`

File: staff/daria-potemkina/web-components/my-app-react/logic/tweet.spec.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+                                done()
+                        })
+                    
+                    })
+                })
+            })
+
+
+    it('should fail if invalid token is introduced', done => {
+        let __token = 'aaaaaaaaaaaa'
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/logic/tweet.spec.js:70]: you don't need to use let here

File: staff/daria-potemkina/web-components/my-app-react/main.jsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
+ReactDOM.render(<App />, document.getElementById('root'))
+
+// const landing = new Landing(
```
Comment (lichab) [inline staff/daria-potemkina/web-components/my-app-react/main.jsx:3]: This whole unnecesary comment here, make the PR really dirty. Remove it

File: staff/daria-potemkina/web-components/my-app-react/components/App.jsx (hunk 1 above)
Comment (Dashkent) [reply staff/daria-potemkina/web-components/my-app-react/components/App.jsx]: comments removed

File: staff/daria-potemkina/web-components/my-app-react/components/Home.jsx (hunk 2 above)
Comment (Dashkent) [reply staff/daria-potemkina/web-components/my-app-react/components/Home.jsx]: try catch applied in all function for synchronous errors. I have used throw error.

**Pull request merged**